ballerina/password.bal

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) 2025 WSO2 LLC. (http://www.wso2.com).
+//
+// WSO2 LLC. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/jballerina.java;
+
+# Returns a BCrypt hash of the given password with optional work factor.
+# ```ballerina
+# string password = "mySecurePassword123";
+# string|crypto:Error hash = crypto:hashPassword(password);
+# ```
+#
+# + password - Password string to be hashed
+# + workFactor - Optional work factor (cost parameter) between 4 and 31. Default is 12
+# + return - BCrypt hashed password string or Error if hashing fails
+public isolated function hashPassword(string password, int workFactor = 12) returns string|Error = @java:Method {
```

Comment:
```suggestion
public isolated function hashBcrypt(string password, int workFactor = 12) returns string|Error = @java:Method {
```
To allign with the current API naming convention let's change the API to   `hashBcrypt`, `verifyBcrypt`, and `generateBcryptSalt`